tests/yast2_gui/yast2_users.pm

```diff
@@ -16,11 +16,107 @@ use strict;
 use warnings;
 use testapi;
 
+use utils qw(type_string_slow_extended);
+use version_utils qw(is_tumbleweed is_leap);
+
+sub write_change_now {
+    send_key 'alt-x';
+    send_key 'down';
+    send_key 'down';
+    send_key 'down';
+    send_key 'ret';
+}
+
+sub check_with_xterm {
+    my $command = shift;
+    x11_start_program('xterm');
+    become_root;
+    assert_script_run($command);
+    wait_screen_change { send_key 'alt-f4'; };
+}
+
+sub add_user {
+    my (%args) = @_;
+    send_key 'alt-a';    # add user tab
+    type_string_slow_extended($args{username});
+    send_key 'tab';
+    type_string_slow_extended($args{username});
+    send_key 'tab';
+    type_string_slow_extended($args{password});
+    send_key 'tab';
+    type_string_slow_extended($args{password});
+    assert_screen 'user_data_fields_filled_up';
+    send_key "alt-o";    # OK => Exit
+    if (is_leap || is_tumbleweed) {
+        wait_still_screen 3;
+        send_key 'alt-n';
+    }
+    assert_screen 'new_user_has_been_created';
+}
+
+sub edit_user {
+    send_key 'alt-i';    # edit user tab
+    send_key 'tab';      # change user full name
+    send_key 'tab';      # we need 2 tabs
+    type_string_slow_extended('edited');
+    send_key 'tab';      # change username
+    type_string_slow_extended('edited');
+    send_key "alt-o";
+    assert_screen 'home_dir_popup__checking';
+    send_key 'ret';
+    assert_screen 'edited_user_checking';
+}
+
+sub show_user_info {
+    send_key 'alt-i';
+    send_key 'alt-d';
+    assert_screen 'show_user__checking';
+    send_key 'alt-o';
+}
+
+sub delete_user {
+    send_key 'alt-t';
+    assert_screen 'delete_user_popup__checking';
+    send_key 'alt-y';
+}
+
+sub show_help {
+    send_key 'alt-h';
+    assert_screen 'help_popup__checking';
+    send_key 'alt-c';
+}
+
+sub show_system_users {
+    send_key 'alt-s';
+    send_key 'down';
+    send_key 'down';
+    send_key 'ret';
+    assert_screen 'list_users__checking';
```

Comment:
In general, sequencing multiple *send_keys* without any assertion can lead to difficulties for openQA job reviewers in case of failure. Simply because the SUT will lose synchronization with the test code. 
This can easily happen when you will transmit from a tab to another tab in any yast2 module. Usually when SUT is starting any yast2 module, or when there is some tab change, yast2 needs to load default data. This takes some time and yast2 module might be during this data load time irresponsive. So the test code will be sending keys to the SUT, but nothing will happen and since send_key does not provide any return code or assertion by itself, it has to be handed by the test maintainer.